src/pharmpy/tools/external/nonmem/run.py

```diff
@@ -17,16 +17,10 @@
 PARENT_DIR = f'..{os.path.sep}'
 
 
-def execute_model(model_or_model_entry, db):
-    if isinstance(model_or_model_entry, ModelEntry):
-        model = model_or_model_entry.model
-
-        # TODO : Remove after complete integration of ModelEntry
-        if model.modelfit_results is not None:
-            model = model.replace(modelfit_results=None)
-            model_or_model_entry = ModelEntry.create(model, parent=model_or_model_entry.parent)
-    else:
-        model = model_or_model_entry
+def execute_model(model_entry, db):
```

Comment:
Why can't we have model here? And create the `ModelEntry` in the function.

Reply:
The `parent` attribute is set when creating the candidate models in tools, so in order to not have to attach that information after fitting it takes a `ModelEntry`.

Reply:
I guess we could still have support for a `Model` object but it seems a bit counteractive to the `ModelEntry` implementation

Reply:
I agree now to keep this. My thinking was that the `execute_model` was the source of the `ModelEntry` objects. But  `parent` and potentially future other things cannot be reconciled with this point of view.